packages/headless/src/controllers/facet-manager/headless-facet-manager.ts

```diff
@@ -1,18 +1,83 @@
 import {SearchEngine} from '../../app/search-engine/search-engine';
+import {setDesiredCount} from '../../features/facets/automatic-facet-set/automatic-facet-set-actions';
+import {loadAutomaticFacetSetActions} from '../../features/facets/automatic-facet-set/automatic-facet-set-actions-loader';
+import {loadReducerError} from '../../utils/errors';
 import {
   buildCoreFacetManager,
-  FacetManager,
-  FacetManagerState,
+  FacetManager as CoreFacetManager,
+  FacetManagerState as CoreFacetManagerState,
   FacetManagerPayload,
 } from '../core/facet-manager/headless-core-facet-manager';
+import {
+  AutomaticFacet,
+  buildAutomaticFacet,
+} from '../facets/automatic-facet/headless-automatic-facet';
+
+export type {CoreFacetManagerState, FacetManagerPayload, CoreFacetManager};
 
-export type {FacetManagerState, FacetManagerPayload, FacetManager};
+export interface FacetManagerState extends CoreFacetManagerState {
+  /**
+   * @beta - This property is part of the automatic facets feature.
+   * Automatic facets are currently in beta testing and should be available soon.
+   *
+   * The list of automatic facet controllers.
+   */
+  automaticFacets?: AutomaticFacet[];
+}
+
+export interface FacetManager extends CoreFacetManager {
+  /**
+   * The state of the `FacetManager` controller.
+   */
+  state: FacetManagerState;
+}
 
+export interface FacetManagerProps {
+  /**
+   * @beta - This property is part of the automatic facets feature.
+   * Automatic facets are currently in beta testing and should be available soon.
+   *
+   * The desired count of automatic facets.
+   * Must be a positive integer.
+   */
+  desiredCount: number;
+}
 /**
  * Creates a `FacetManager` instance.
  *
  * @param engine - The headless engine.
+ * @param props - The optional facet manager props for the automatic facets feature. Automatic facets are currently in beta testing and should be available soon.
+ * @returns A `FacetManager` controller instance.
  */
-export function buildFacetManager(engine: SearchEngine): FacetManager {
-  return buildCoreFacetManager(engine);
+export function buildFacetManager(
+  engine: SearchEngine,
+  props?: FacetManagerProps
+): FacetManager {
+  if (!props?.desiredCount) {
+    return buildCoreFacetManager(engine);
+  }
+  if (!loadAutomaticFacetSetActions(engine)) {
```

Comment:
You are right. It makes no sense. I added the `loadFacetManagerReducers` it to copy how other controllers do it. Eventually, it will have more reducers.